app/helpers/application_helper/button/general_image.rb

```diff
@@ -0,0 +1,7 @@
+class ApplicationHelper::Button::GeneralImage < ApplicationHelper::Button::Basic
```

Comment:
I'm not so sure how you use `Basic` in this context. I would assume it is something like a `BaseClass`, but with a concrete implementation, which also gets instantiated and used somewhere. So, if `GeneralImage` is used somewhere, but also might be extended, I think `BasicImage` is better, because its the same name and concept as in `Toolbar::Basic` and `Button::Basic`

ps. I don't see any instantiation of `Toolbar::Basic`, but only inheritance. If you use it more as a Base class, you might consider renaming that `Toolbar::Base` :stuck_out_tongue_winking_eye: